### PR TITLE
Move e2e ingress test manifests for GCE to a subdirectory

### LINKED ISSUE #58640: Manifests in test/e2e/testing-manifests/ingress related to GCE should be in a separate folder.
Manifests should ideally be in a folder gce/ for clarity.

/assign

### PR DESCRIPTION
**Pull request by vllry**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The GCE-specific ingress manifest resources were organized inconsistently compared to the nginx-specific ones (which live in a subdirectory). This PR moves the GCE `static-ip-2` assets to a `gce` directory, and updates path references.

**Which issue(s) this PR fixes**:

Fixes #58640 

**Special notes for your reviewer**:
This is my first kubernetes/kubernetes PR, so please let me know if I've made an obvious or beginner's mistake.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
/retest
TestTerminalPodEviction flakes strike again! :(

**Comment by vllry** [comment]
Tests were indeed a flake, just need lgtm + approval. 😁

**Comment by oomichi** [comment]
Manifests of static-ip-2 are used only for e2e tests of gce/gke. So

/lgtm

**Comment by vllry** [comment]
@thockin @oomichi @eparis This PR is pretty old + small, would appreciate a review (or if you could toss in a different reviewer)

File: test/e2e/upgrades/ingress.go [hunk 1]
```diff
@@ -95,10 +95,9 @@ func (t *IngressUpgradeTest) Setup(f *framework.Framework) {
 	// Allocate a static-ip for the Ingress, this IP is cleaned up via CleanupGCEIngressController
 	t.ipName = fmt.Sprintf("%s-static-ip", ns.Name)
 	t.ip = t.gceController.CreateStaticIP(t.ipName)
-
```
Comment (oomichi) [inline test/e2e/upgrades/ingress.go]: This seems unnecessary change here for readability.

**Review by oomichi — COMMENTED** [review]
@vllry Thanks for rebasing, just one comment

File: test/e2e/upgrades/ingress.go (hunk 1 above)
Comment (vllry) [reply test/e2e/upgrades/ingress.go]: 👍

**Comment by oomichi** [comment]
Thanks for updating,

/lgtm

**Comment by vllry** [comment]
Thanks, still need a `/approve`.

**Comment by vllry** [comment]
@oomichi @eparis

**Comment by oomichi** [comment]
I don't have power of approval.
@spiffxp Could you take a look at this?
